src/main/java/org/hydev/mcpm/client/database/SuperLocalPluginTracker.java

```diff
@@ -0,0 +1,576 @@
+package org.hydev.mcpm.client.database;
+
+import com.opencsv.CSVReader;
+import com.opencsv.CSVWriter;
+import com.opencsv.exceptions.CsvException;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.hydev.mcpm.client.database.boundary.SearchPackagesBoundary;
+import org.hydev.mcpm.client.database.inputs.SearchPackagesInput;
+import org.hydev.mcpm.client.database.inputs.SearchPackagesType;
+import org.hydev.mcpm.client.database.results.SearchPackagesResult;
+import org.hydev.mcpm.client.models.PluginModel;
+import org.hydev.mcpm.client.models.PluginVersion;
+import org.hydev.mcpm.client.models.PluginYml;
+import org.hydev.mcpm.utils.PluginJarFile;
+
+import java.io.*;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * 
+ * Representation of a plugin for use in the local plugin tracker.
+ */
+
+class PluginTrackerModel {
+    String name;
+    Boolean isManual;
+    String versionId;
+    String pluginId;
+
+    public PluginTrackerModel(String name, Boolean isManual, String versionId, String pluginId) {
+        this.name = name;
+        this.isManual = isManual;
+        this.versionId = versionId;
+        this.pluginId = pluginId;
+    }
```

Comment:
Please don't put two classes in one file, and also you can use records instead of classes to reduce the boilerplate like

```java
public record PluginTrackerModel(
    String name,
    boolean isManual,
    ...
) { }
```